plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -140,14 +225,16 @@ public function payment_complete( $transaction_id = '' ) {
 				WC()->session->set( 'order_awaiting_payment', false );
 			}
 
-			if ( $this->has_status( apply_filters( 'woocommerce_valid_order_statuses_for_payment_complete', array( 'on-hold', 'pending', 'failed', 'cancelled' ), $this ) ) ) {
+			// phpcs:disable WooCommerce.Commenting.CommentHooks.MissingHookComment
```

Comment:
I'm curious about this one, but let's consider other comments first.

Reply:
I had to add this PHPCS comment because I changed this line when I replaced strings with the constants. Apparently, the rule was added after this line.

Reply:
So I know it means a little extra work 😬 ... but if we change a line we should generally make it compliant with current standards. 

Of the three options (adding a hook comment, ignoring and making an exception, and suppressing the rule), I think suppressing is the least attractive option as it increases the chances that this will also be skipped if, for example, additional parameters are added to the hook at some future point.

Reply:
Makes sense! I have included the required comments now 👍